src/System.Net.Requests/src/System/Net/HttpWebRequest.cs

```diff
@@ -91,6 +95,55 @@ private enum Booleans : uint
             IsWebSocketRequest = 0x00000800,
             Default = AllowAutoRedirect | AllowWriteStreamBuffering | ExpectContinue
         }
+
+        private class HttpClientParameters
+        {
+            public readonly DecompressionMethods AutomaticDecompression;
+
+            public readonly bool AllowAutoRedirect;
+
+            public readonly int MaximumAutomaticRedirections;
+
+            public readonly int MaximumResponseHeadersLength;
+
+            public readonly bool PreAuthenticate;
+
+            public readonly int Timeout;
+
+            public readonly SslProtocols SslProtocols;
+
+            public readonly bool CheckCertificateRevocationList;
+
+            public HttpClientParameters(HttpWebRequest webRequest)
+            {
+                AutomaticDecompression = webRequest.AutomaticDecompression;
+                AllowAutoRedirect = webRequest.AllowAutoRedirect;
+                MaximumAutomaticRedirections = webRequest.MaximumAutomaticRedirections;
+                MaximumResponseHeadersLength = webRequest.MaximumResponseHeadersLength;
+                PreAuthenticate = webRequest.PreAuthenticate;
+                Timeout = webRequest.Timeout;
+                SslProtocols = (SslProtocols)ServicePointManager.SecurityProtocol;
+                CheckCertificateRevocationList = ServicePointManager.CheckCertificateRevocationList;
+            }
+
+            public bool Matches(HttpWebRequest webRequest)
+            {
+                return AutomaticDecompression == webRequest.AutomaticDecompression && AllowAutoRedirect == webRequest.AllowAutoRedirect
+                    && MaximumAutomaticRedirections == webRequest.MaximumAutomaticRedirections
+                    && MaximumResponseHeadersLength == webRequest.MaximumResponseHeadersLength && PreAuthenticate == webRequest.PreAuthenticate
+                    && Timeout == webRequest.Timeout && SslProtocols == (SslProtocols)ServicePointManager.SecurityProtocol
+                    && CheckCertificateRevocationList == ServicePointManager.CheckCertificateRevocationList;
+            }
+
+            public static bool AreParametersAcceptableForCaching(HttpWebRequest webRequest)
+            {
+                return webRequest._credentials == null && ReferenceEquals(webRequest._proxy, DefaultWebProxy)
+                    && webRequest.ServerCertificateValidationCallback == null && ServicePointManager.ServerCertificateValidationCallback == null
```

Comment:
Nit: this formatting, in both this function and the one above, is making it more difficult to see everything that's being checked.  Can you please reformat these to have one comparison per line?